### PR TITLE
Slightly faster method for newton iteration in `E` and `H` calculation

### PR DESCRIPTION
**Pull request by Yash-10**
Hello, as per #1240, this is an attempt to slightly make the computation faster (even thought both functions were already jitted).

Thanks!

### CODE REVIEW

**Comment by Yash-10** [comment]
As was seen locally, there are slight mismatches. The eccentricity conditions in the hyperbolic case might need to be investigated...

File: src/poliastro/core/angles.py [hunk 1]
```diff
@@ -311,7 +310,16 @@ def M_to_F(M, ecc):
     This uses a Newton iteration on the hyperbolic Kepler equation.
 
     """
-    F0 = np.arcsinh(M / ecc)
+    if ecc < 1.25:
+        if -np.pi < M < 0 or M > np.pi:
```
Comment (astrojuanlu) [inline src/poliastro/core/angles.py:314]: My recommendation: "Always use less-than" https://thehardcorecoder.com/2014/03/05/always-use-less-than/ :)

**Comment by astrojuanlu** [comment]
> although of magntiude ~1e-20 or lower

Where exactly was this error appearing? I think it should be acceptable to adapt the test accordingly

Also, could you show a comparison of speed or number of iterations before and after?

File: src/poliastro/core/angles.py (hunk 1 above)
Comment (Yash-10) [reply src/poliastro/core/angles.py]: Thanks! That's interesting to know :)

**Comment by Yash-10** [comment]
> Also, could you show a comparison of speed or number of iterations before and after?

I haven't changed the number of iterations for the newton iteration. For speed improvements, there's a slight improvement. I had done an analysis in #1240 for discussion. Maybe that would give an idea?


> The functions are jitted and using the timeit module gives the following rough analysis:
Note: the below "no. of iterations" are the no. of times both the implementations are run using timeit and not the no. of iterations for the newton-raphson method.

> - For 1000 iterations, there is ~0.26 s time improvement.
> - For 10000 iterations, there is ~2.62 s time improvement.
> - For 100000 iterations, there is ~26.08 s improvement.
> Summarizing it in a graph:

> image

**Comment by Yash-10** [comment]
As you say, we could make the implementation look exactly like Vallado's (i.e. change the eccentricty condition to < 1.6 in the hyperbolic case) and modify the tests to prevent any minor errors.

**Comment by astrojuanlu** [comment]
Yes, let's do it!

**Review by astrojuanlu — APPROVED** [review]
Thanks @Yash-10 !

**Pull request merged**